packages/yourloops/lib/team/team-api.ts

```diff
@@ -62,9 +66,20 @@ interface RemoveMemberArgs {
 }
 
 export default class TeamApi {
-  static async getTeams(): Promise<ITeam[]> {
+  static getTeamsApiUrl(user: User): string {
+    if (user.isUserHcp()) {
+      return `/bff/v1/hcps/${user.id}/teams`
+    } else if (user.isUserPatient()) {
+      return `/bff/v1/patients/${user.id}/teams`
+    } else {
+      throw Error(`User with role ${user.role} cannot retrieve teams`)
+    }
```

Comment:
Some improvements could be done to this method:
- The error case could be handled first
- The HCP and patient case could be factorized

```ts
// To put at the top of the file to make sure these `const` never change
const HCP_ROUTE = 'hcps'
const PATIENTS_ROUTE = 'patients'

// ...
static getTeamsApiUrl(user: User): string {
  if (!user.isUserHcp() && !user.isUserPatient() {
    throw Error('...')
  }
  const userRoute = user.isUserHcp() ? HCP_ROUTE : PATIENT_ROUTE
  return `/bff/v1/${userRoute}/${user.id}/teams`
}

```